x-pack/plugins/enterprise_search/server/lib/ml_inference_pipeline/get_inference_pipelines.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core/server';
+import { MlTrainedModels } from '@kbn/ml-plugin/server';
+
+import { IngestPipeline, IngestProcessorContainer } from '@elastic/elasticsearch/lib/api/types';
+
+/**
+ * Gets all ML inference pipelines. Redacts trained model IDs in those pipelines which reference
+ * a model inaccessible in the current Kibana space.
+ * @param esClient the Elasticsearch Client to use to fetch the errors.
+ * @param trainedModelsProvider ML trained models provider.
+ */
+export const getMlInferencePipelines = async (
+  esClient: ElasticsearchClient,
+  trainedModelsProvider: MlTrainedModels | undefined,
+): Promise<Record<string, IngestPipeline>> => {
+  if (!trainedModelsProvider) {
+    return Promise.reject(new Error('Machine Learning is not enabled'));
+  }
+
+  // Fetch all ML inference pipelines
+  const fetchedInferencePipelines = await esClient.ingest.getPipeline({
+    id: 'ml-inference-*',
+  });
+
+  // Fetch all trained model IDs that are accessible in the current Kibana space
+  const trainedModels = await trainedModelsProvider.getTrainedModels({});
+  const accessibleModelIds = Object.values(trainedModels.trained_model_configs)
+    .map((modelConfig) => modelConfig.model_id);
+
+  // Process pipelines: check if the model_id is one of the redacted ones, if so, redact it in the
+  // result as well
+  const inferencePipelinesResult: Record<string, IngestPipeline> = {};
+  Object.keys(fetchedInferencePipelines).forEach((name) => {
```

Comment:
Using [Object.entries](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries) instead of `Object.keys` would be a little more concise here.

```typescript
Object.entries(fetchedInferencePipelines).forEach(([name, inferencePipeline]) => {});
```